src/efield_tb_methods.F

```diff
@@ -91,7 +91,12 @@ SUBROUTINE efield_tb_matrix(qs_env, ks_matrix, rho, mcharge, energy, calculate_f
       energy%efield = 0.0_dp
       CALL get_qs_env(qs_env, dft_control=dft_control)
       IF (dft_control%qs_control%dftb .OR. dft_control%qs_control%xtb) THEN
-         IF (dft_control%apply_period_efield) THEN
+         if_apply: IF (dft_control%apply_period_efield) THEN
+            ! check if the periodic efield should be applied in the current step
+            IF (dft_control%period_efield%start_frame > qs_env%sim_step) EXIT if_apply
```

Comment:
I didn't realize one can use `EXIT` on if-blocks. This smells a lot like GOTO. A better solution would be to use something like this: `IF... ELSE IF...ELSE IF... ELSE..END IF`.

Reply:
I fixed that, thanks for noticing